core/templates/dev/head/pages/exploration_player/TutorCardDirective.js

```diff
@@ -0,0 +1,140 @@
+
+    // Fix the current card height, so that it does not change during the
+    // animation, even though its contents might.
+    element.css('height', currentHeight);
+
+    jQuery(element).animate({
+      opacity: 0
+    }, TIME_FADEOUT_MSEC).animate({
+      height: expectedNextHeight
+    }, TIME_HEIGHT_CHANGE_MSEC).animate({
+      opacity: 1
+    }, TIME_FADEIN_MSEC, function() {
+      element.css('height', '');
+      done();
+    });
+
+    return function(cancel) {
+      if (cancel) {
+        element.css('opacity', '1.0');
+        element.css('height', '');
+        element.stop();
+      }
+    };
+  };
+
+  return {
+    addClass: animateCardChange
+  };
+});
+
+oppia.directive('tutorCard', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      startCardChangeAnimation: '=',
+      showNextCard: '&',
+      showSupplementalCard: '&'
+    },
+    templateUrl: 'components/TutorCard',
+    controller: [
+    '$scope', 'oppiaPlayerService', 'UrlInterpolationService',
+    'playerPositionService', 'playerTranscriptService',
+    'ExplorationObjectFactory', 'windowDimensionsService',
+    function($scope, oppiaPlayerService, UrlInterpolationService,
+      playerPositionService, playerTranscriptService, explorationObjectFactory,
+      windowDimensionsService) {
+      var CONTENT_FOCUS_LABEL_PREFIX = 'content-focus-label-';
+      var TWO_CARD_THRESHOLD_PX = 960;
+
+      var updateActiveCard = function() {
+        var index = playerPositionService.getActiveCardIndex();
+        if (index === null) {
+          return;
+        }
+
+        var exploration = explorationObjectFactory.get();
+        $scope.arePreviousResponsesShown = false;
+        $scope.activeCard = playerTranscriptService.getCard(index);
+        $scope.isInteractionInline = exploration.isInteractionInline(
+          $scope.activeCard.stateName);
+        $scope.interactionInstructions = exploration.getInteractionInstructions(
+          $scope.activeCard.stateName);
+      };
+
+      $scope.arePreviousResponsesShown = false;
+
+      $scope.OPPIA_AVATAR_IMAGE_URL = (
+        UrlInterpolationService.getStaticImageUrl(
+          '/avatar/oppia_black_72px.png'));
+
+      $scope.profilePicture = (UrlInterpolationService.getStaticImageUrl(
+          '/avatar/user_blue_72px.png'));
+
+      oppiaPlayerService.getUserProfileImage().then(function(result) {
+        $scope.profilePicture = result;
+      });
+
+      $scope.getContentFocusLabel = function(index) {
+        return CONTENT_FOCUS_LABEL_PREFIX + index;
+      };
+
+      $scope.toggleShowPreviousResponses = function() {
+        $scope.arePreviousResponsesShown = !$scope.arePreviousResponsesShown;
+      };
+
+      $scope.isViewportNarrow = function() {
+        return windowDimensionsService.getWidth() < TWO_CARD_THRESHOLD_PX;
+      };
+
+      $scope.submitAnswer = function(answer, interactionRulesService) {
+        $scope.$parent.submitAnswer(answer, interactionRulesService);
```

Comment:
Never use $parent. Pass the function in if you need to, or call a service directly?